File: Problem 014/Solution 14.cpp
#include <iostream>

using std::cout;
using std::endl;

int Even(int number)
{
	return number/2;
}

int Odd(int number)
{
	return (number*3)+1;
}

void StartChain(int number, int& Largest, int& LastCount)
{
	/* counter is the current count */
	int counter = 0;
	int Current = number;

	while (number != 1)
	{
		if ((number % 2) == 0)
		{
			number = Even(number);
		}
		else
		{
			number = Odd(number);
		}
		counter++;
	}

	if (counter > LastCount)
	{
		Largest = Current;
	}

}

int main()
{
	int Largest = 0;
	int i = 1000000;
	int LastCount = 0;
	for(i = 1000000; i > 0; i--)
	{
		StartChain(i, Largest, LastCount);
	}

	cout << "Largest chain was on: " << Largest << endl;

	return 0;
}

File: Problem 014/Solution 14.rb
# Solves http://projecteuler.net/problem=14

def Even(number)
	return number/2
end

def Odd(number)
	return 3*number+1
end


l_number = 0
l_counter = 0

for i in (1..1000000)
	check = i
	counter = 0

	while check != 1
		if (check % 2 == 0)
			check = Even(check)
		else
			check = Odd(check)
		end

		counter += 1
	end

	if (l_counter < counter)
		l_number = i
		l_counter = counter
	end
end

print "Answer: ", l_number, "\n"
